pkg/webhook/patch.go

```diff
@@ -299,13 +297,13 @@ func addTolerations(pod *corev1.Pod, app *v1beta1.SparkApplication) []patchOpera
 
 func addSchedulerName(pod *corev1.Pod, app *v1beta1.SparkApplication) *patchOperation {
 	var schedulerName *string
-	if util.IsDriverPod(pod) {
+	if util.IsDriverPod(pod) && app.Spec.Driver.SchedulerName != nil {
```

Comment:
You don't need the additional condition here since you check `schedulerName == nil` below.